### ISSUE TITLE
Can't compile angular 8 under typescript "~3.5.3"

### ISSUE THREAD

**Issue by kilatib** [ui-router/core#507]
Environment:    
    "@angular/common": "~8.2.14",
    "@angular/compiler": "~8.2.14",
    "@angular/core": "~8.2.14",
    "@uirouter/angular": "^6.0.1",
    "@uirouter/angular-hybrid": "^10.0.1",
    "@uirouter/angularjs": "^1.0.23",
    "@uirouter/core": "^6.0.2",
    "@uirouter/rx": "^0.6.4",
    "typescript": "~3.5.3"

Command: **ng build --prod**

> ERROR in ../node_modules/@uirouter/core/lib/url/urlRouter.d.ts:71:9 - error TS1086: An accessor cannot be declared in an ambient context.

>71     get interceptDeferred(): boolean;
           ~~~~~~~~~~~~~~~~~
>../node_modules/@uirouter/core/lib/state/stateService.d.ts:26:9 - error TS1086: An accessor cannot be declared in an ambient context.

>26     get transition(): Transition;
           ~~~~~~~~~~
>../node_modules/@uirouter/core/lib/state/stateService.d.ts:32:9 - error TS1086: An accessor cannot be declared in an ambient context.

>32     get params(): StateParams;
           ~~~~~~
>../node_modules/@uirouter/core/lib/state/stateService.d.ts:38:9 - error TS1086: An accessor cannot be declared in an ambient context.

>38     get current(): StateDeclaration;
           ~~~~~~~
>../node_modules/@uirouter/core/lib/state/stateService.d.ts:44:9 - error TS1086: An accessor cannot be declared in an ambient context.

>44     get $current(): StateObject;
           ~~~~~~~~

**Comment by kilatib** [comment]
Fix to me added to posinstall script: 

 

>  (function(){
>     const fileList = [
>       __dirname + '/node_modules/@uirouter/core/lib//url/urlRouter.d.ts',
>       __dirname + '/node_modules/@uirouter/core/lib/state/stateService.d.ts'
>     ];
> 
>     fileList.forEach((file) => {
>       fs.readFile(file, 'utf8', function (err,data) {
>         if (err) {
>           return console.log(err);
>         }
>         var result = data.replace(/get\s([^(]+)\(\)/g, '$1');
> 
>         fs.writeFile(file, result, 'utf8', (err) => console.log(err));
>       });
>     });
>   })();

**Comment by alandeg** [comment]
I'm getting the exact same issue

**Comment by christopherthielen** [comment]
This was fixed in core v6.0.3 
https://github.com/ui-router/core/releases/tag/6.0.3
